### PR TITLE
Working client - server 

### PR DESCRIPTION
**Pull request by matipatapon**
I used regex for getting values from request but I didn't think it would be so slow ... I need to refactor it.

### CODE REVIEW

File: src/Common/loginfo.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <windows.h>
+#include <QString>
+
+struct LogInfo{
+    LogInfo(const DWORD pid);
+    QString filePath = "";
```
Comment (ElwooD07) [inline src/Common/loginfo.h:7]: QString is empty by default

File: src/Common/loginfo.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    QString operationType = "";
+    QString preview = "";
+    QString fileHandle = "";
+    QString resultOfTheOperation = "";
+    DWORD pid;
```
Comment (ElwooD07) [inline src/Common/loginfo.h:12]: Give it default value 0

File: src/Common/pipeclient.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+class PipeClient : public PipeHost
+{
+public:
+    static bool receivePermission(const DWORD Pid, char& permission);
+    static bool sendLog(const LogInfo& logInfo);
+    static std::string parseLogInfoIntoRequest(const LogInfo& logInfo);
+private:
+    static HANDLE openPipe(LPCWSTR pineName = PipeName);
```
Comment (ElwooD07) [inline src/Common/pipeclient.h]: Open pipe from pipehost and handle it by ATL::CHandle, and give the ability to derived classes to use it (make it protected)

**Review by ElwooD07 — APPROVED** [review]
I have no time to finish the review. Please make sure that all you have implemented is working fine before the presentation.

**Review by ccpp-apriorit — APPROVED** [review]
I have no time to finish the review, please read the comments from ElwooD07

**Pull request merged**